### ISSUE TITLE
`:test` does not find any tests in REPL

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#6137]

**Comment by magnus-madsen** [comment]
Starting point is probably in LanguageServer

**Comment by sockmaster27** [comment]
It looks like everything in Bootstrap.scala is broken

**Comment by sockmaster27** [comment]
Except for `:build`, apparently because it takes `flix: Flix` as a parameter, while all the others construct a new one themselves. Is this intentional?

**Comment by sockmaster27** [comment]
Weird - Passing the REPL's `flix` to `Bootstrap.test` works until you evaluate anything else. Is this a different bug, or am I misunderstanding something?

**Comment by magnus-madsen** [comment]
Hmm. Its sounds complicated. Maybe we need Jonathan or Matt to look into it. Thanks for the initial debugging.

**Comment by emk** [comment]
I can confirm that this is happening in Flix 0.42.0 under VS Code on Linux. Please let me know if there's anything I can do to help track this down!

**Comment by JonathanStarup** [comment]
@emk thank you for pushing me to fixing this finally. I'm not sure I'm gonna fix it today so just for public record, it works if you remove execReload in Shell.scala. Ill keep digging

**Comment by emk** [comment]
Ah, thank you!

I was experimenting with Flix to sketch out algorithms for two open source projects at work:

- [A BigQuery transpiler](https://github.com/faradayio/joinery), which does some messy type inference over a lattice of types. (The full BigQuery type system is... odd.)
- [dbcrossbar](https://www.dbcrossbar.org/), which will soon need to calculate least-cost-paths over different kinds of tabluar data conversions (some of them parametric).

I've been using Prolog for some of these algorithm sketches, but Datalog is cleaner. And the lattice fixpoint stuff is a feature I never even imagined I needed until I saw it. :-)

I'd love to write up a blog post talking about at least one of these at some point. If you'd like, I'll try to report any rough edges I hit as brand new user.

**Comment by JonathanStarup** [comment]
That sounds very cool! feel free to submit your issues yes, it is valuable feedback

**Comment by JonathanStarup** [comment]
@magnus-madsen The issues is this code in `Bootstrap` where a new `Flix` instance is created with the old settings but clearing all sources. This causes the issue because `reconfigureFlix` only adds changed files. this means that all non-new files are removed essentially. I think the solution is for someone to look over the usage of `Flix` in `Bootstrap`. (it holds paths that relate to a specific flix instance but doesn't keep track of that instance itself? that seems suspicious)
```scala
  def test(o: Options): Validation[Unit, BootstrapError] = {
    implicit val flix: Flix = new Flix().setFormatter(Formatter.getDefault)
    flix.setOptions(o)
    flatMapN(build()) {
      compilationResult =>
        Tester.run(Nil, compilationResult) match {
          case Ok(_) => ().toSuccess
          case Err(_) => Validation.toHardFailure(BootstrapError.GeneralError(List("Tester Error")))
        }
    }
  }
```

**Comment by magnus-madsen** [comment]
@JonathanStarup 

Perhaps ´benchmark`, `doc`, `run`, and `test` should all take a Flix object explicitly?

**Comment by JonathanStarup** [comment]
> @JonathanStarup 
> 
> Perhaps `benchmark`, `doc`, `run`, and `test` should all take a Flix object explicitly?

yeah something like that, but i also think Bootstrap should hold a flix object since its memorizing paths for some implied flix

**Comment by magnus-madsen** [comment]
I have posted a fix here #6871

**Comment by magnus-madsen** [comment]
Fixed-- but confirmation would be nice.
